apps/explorer/src/components/module/PkgModulesWrapper.tsx

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Combobox } from '@headlessui/react';
+import clsx from 'clsx';
+import { useState, useEffect, useCallback } from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+import ModuleView from './ModuleView';
+
+import { ReactComponent as SearchIcon } from '~/assets/SVGIcons/24px/Search.svg';
+import { Tab, TabGroup, TabList, TabPanel, TabPanels } from '~/ui/Tabs';
+import { ListItem, VerticalList } from '~/ui/VerticalList';
+
+type ModuleType = [moduleName: string, code: string];
+
+interface Props {
+    id?: string;
+    modules: ModuleType[];
+}
+
+interface ModuleViewWrapperProps {
+    id?: string;
+    selectedModuleName: string;
+    modules: ModuleType[];
+}
+
+const initialSelectModule = (searchParams: any, modulenames: string[]) => {
+    const paramModule = searchParams.get('module') || modulenames?.[0] || null;
+
+    if (!!paramModule && modulenames.includes(paramModule)) {
+        return paramModule;
+    } else {
+        return modulenames[0];
+    }
+};
+
+function ModuleViewWrapper({
+    id,
+    selectedModuleName,
+    modules,
+}: ModuleViewWrapperProps) {
+    const selectedModuleData = modules.find(
+        ([name]) => name === selectedModuleName
+    );
+
+    if (selectedModuleData) {
+        const [name, code] = selectedModuleData;
+
+        return <ModuleView id={id} name={name} code={code} />;
+    }
+
+    return <div />;
+}
+
+function PkgModuleViewWrapper({ id, modules }: Props) {
+    const modulenames = modules.map(([name], idx) => name);
+    const [searchParams, setSearchParams] = useSearchParams();
+    const [query, setQuery] = useState('');
+    const [selectedModule, setSelectedModule] = useState(
+        initialSelectModule(searchParams, modulenames)
+    );
+
+    const filteredModules =
+        query === ''
+            ? modulenames
+            : modules
+                  .filter(([name]) => name.startsWith(query))
+                  .map(([name]) => name);
+
+    useEffect(() => {
+        setSearchParams({ module: selectedModule });
```

Comment:
You'll need to construct the new search params using the old search params. You can see an example here: https://github.com/MystenLabs/sui/blob/ea8528c06864adfd45673da4ef93b66061286c41/apps/explorer/src/components/transaction-card/RecentTxCard.tsx#L139

Reply:
The code as-is is using the in-built `useSearchParams` hook provided by `react-route-dom`. The documentation suggests that this hook can be used to change the search params. Testing this functionality in-browser seems to support the documentation's assertion. Is this incorrect?

The documentation is available here: https://reactrouter.com/en/main/hooks/use-search-params

Reply:
The issue is that the API doesn’t merge search params by default, which we need to do to preserve the currently selected network.